### DISCUSSION TITLE
Running multiple instance and bind with different search_shortcut

### DISCUSSION THREAD

**Discussion by dddping** [espanso/espanso#2143] [Feature requests and Ideas]
I wonder anyone want to run multiple instance of espanso, and allow each instance have individual match and search_shortcut? I just do an little experience to tweak the pipe name at _**espanso-ipc\src\lib.rs**_, for line `#[cfg(target_os = "windows")]`. I found that espanso support multiple instance under windows 11 **!**

Breaking the "match" is easier for searching, for example I want to type cc for concat, but concat for pandas and numpy is different, so I can create two espanso instance, one instance give me `pd.concat`, other give me `np.concatenate` (because Python is weak type, IDE don't give you function search).

Anyone agree this feature is useful? It is supported naturally by design?

Here is the code I tweak at **_espanso-ipc\src\lib.rs_**
```rust
#[cfg(target_os = "windows")]
pub fn server<Event: Send + Sync + DeserializeOwned + Serialize>(
  id: &str,
  _: &Path,
) -> Result<impl IPCServer<Event>> {
  let val = env::var("PIPENAMESUFFIX").unwrap();
  let idx = id.to_string() + val.as_str();
  let server = windows::WinIPCServer::new(idx.as_str())?;
  Ok(server)
}

#[cfg(target_os = "windows")]
pub fn client<Event: Serialize + DeserializeOwned>(
  id: &str,
  _: &Path,
) -> Result<impl IPCClient<Event>> {
  let val = env::var("PIPENAMESUFFIX").unwrap();
  let idx = id.to_string() + val.as_str();
  let client = windows::WinIPCClient::new(idx.as_str())?;
  Ok(client)
}
```

I download release version espanso-2.2.1, change the code, build it, and I can successfully run two instance of espanso now.

**Comment by smeech** [comment]
The standard workaround is to have app-specific configurations for different environments. 
One could disable/enable groups of triggers (by renaming their `.yml` files with an underscore `_` prefix), but it's clunky, although I suppose enabling and disabling multiple instances of Espanso is no worse!

**Comment by dddping** [comment]
I believe multiple instance is helpful for heavy user of espanso. I give one more example, you can split contact list by company, and bind search_shortcut Alt+P to company A, and Alt+O to company B. Because there are so many people call Peter, having multiple instance is easier to get the right content. Anyway, it is just a suggestion and I can do this by rebuild espanso myself.
